javascript/app-model-exposure/src/components/DefinitionForm.hooks.tsx

```diff
@@ -61,7 +61,15 @@ export function useDefinitionItems(
 		onDismissEdit,
 	)
 	const items = useMemo(() => {
-		return definitions.map(item => {
+		const primaryIndex = definitions.findIndex(
+			d => d.level === CausalityLevel.Primary,
+		)
+		let sortedDefinitions = [...definitions]
+		if (primaryIndex > 0) {
+			sortedDefinitions.splice(primaryIndex, 1)
```

Comment:
Let's move the array splicing into a generic array utils, and call this `moveElement(ary: T[], sourceIndex: number, targetIndex = 0): T[]`